store/sqlstore/plugin_store.go

```diff
@@ -56,23 +57,23 @@ func (ps SqlPluginStore) SaveOrUpdate(kv *model.PluginKeyValue) (*model.PluginKe
 		// Unfortunately PostgreSQL pre-9.5 does not have an atomic upsert, so we use
 		// separate update and insert queries to accomplish our upsert
 		if rowsAffected, err := ps.GetMaster().Update(kv); err != nil {
-			return nil, model.NewAppError("SqlPluginStore.SaveOrUpdate", "store.sql_plugin_store.save.app_error", nil, err.Error(), http.StatusInternalServerError)
+			return nil, errors.Wrap(err, "failed to update PluginKeyValue")
 		} else if rowsAffected == 0 {
 			// No rows were affected by the update, so let's try an insert
 			if err := ps.GetMaster().Insert(kv); err != nil {
-				return nil, model.NewAppError("SqlPluginStore.SaveOrUpdate", "store.sql_plugin_store.save.app_error", nil, err.Error(), http.StatusBadRequest)
+				return nil, store.NewErrInvalidInput("PluginKeyValue", "<PluginId, Key>", fmt.Sprintf("<%s, %s>", kv.PluginId, kv.Key))
```

Comment:
I'm not sure if this is the type of error that is happening here. In theory the kv is valid, and there wasn't a valid update (so the key wasn't store) looks like this error only happens when the upsert (In this case a external implementation of upsert) fails. So for me, the error should ben the same that we see in the `mysql` case.

Reply:
yep I understand and it makes sense but I think this PR is not intended to fix this, but if you both agree with that suggestion, I will make the correction in another PR later :)

Reply:
Agree with Jesus that this needs to be an internal error. 0/5 on whether we want to do it now or later. Up to you.

Reply:
we are ok with this now :)